src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -358,20 +377,22 @@ def evaluate(
         condition_evaluation, new_priority = self._evaluation_detector_conditions(value)
         state = self.state_manager.get_state_data([None])[None]
 
-        if state.status == new_priority or not condition_evaluation:
+        if (state.status == new_priority) or (  # The status hasn't changed
+            not condition_evaluation
+        ):  # If the condition is not met
             return None
 
-        # TODO - enqueue state update here
-        # TODO - enqueue threshold update
-        # self.state_manager.enqueue_threshold_update(None, new_priority)
+        updated_status_count = (state.counter_updates.get(new_priority) or 0) + 1
+        self.state_manager.enqueue_counter_update(None, {new_priority: updated_status_count})
+
+        if self._thresholds[new_priority] > updated_status_count:
+            # We haven't met the threshold yet, so don't trigger
+            return None
```

Comment:
One thing to keep in mind - probably, higher priorities should also increment counts for all lower priorities.

For metric alerts, if we had a threshold of 3, and then we got `crit, crit, warn`, the warning should fire.

Reply:
🤔 makes sense. will update.